notdiamond/toolkit/custom_router.py

```diff
@@ -80,7 +80,7 @@ def _request_train_router(
 
     def _prepare_joint_dataset(
         self,
-        dataset: Dict[str, pd.DataFrame],
+        dataset: Dict[str | LLMConfig, pd.DataFrame],
```

Comment:
In the interest of backward compatibility with python<3.10, can we instead use `Union`?
```suggestion
        from typing import Union
        dataset: Dict[Union[str, LLMConfig], pd.DataFrame],
```

Reply:
That's a great catch, I hate having to change them later. Thanks